force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,64 @@
+trigger OpportunityTrigger on Opportunity (before update, before delete) {
+
+    // Handle 'before update' context for Opportunity Amount validation
+    if (Trigger.isUpdate) {
+        for (Opportunity oppBefore : Trigger.new) {
+            if (oppBefore.Amount != null && oppBefore.Amount <= 5000) {
+                oppBefore.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+    }
+
+    // Handle 'before delete' context to prevent deletion of Closed Won opportunities for banking accounts
+    if (Trigger.isDelete) {
+        // Set to collect Account IDs of opportunities being deleted
+        Set<Id> accountIds = new Set<Id>();
+
+        // Collect Account IDs from Closed Won opportunities
+        for (Opportunity oppDelete : Trigger.old) {
+            if (oppDelete.StageName == 'Closed Won') {
+                accountIds.add(oppDelete.AccountId);
+            }
+        }
+
+        
+        Map<Id, Account> accountMap = new Map<Id, Account>(
+            [SELECT Id, Industry FROM Account WHERE Id IN :accountIds AND Industry = 'Banking']
+        );
```

Comment:
You can also query the opportunities into a Map easily using the old key values and then use that to check via the Account.Industry relationship field. It will reduce your separate loops and maps.